preview/Dockerfile

```diff
@@ -7,6 +7,16 @@ RUN mkdir -p /run/nginx
 RUN git config --global pack.windowMemory 500m
 RUN git config --global pack.deltaCacheSize 10m
 RUN git config --global pack.threads 1
+# Prevent git auto-gc from warning about too many references by increasing the
+# threshold of unreachable objects that triggers a prune and more aggressively
+# pruning those unreachable objects. While we're at it we more aggressively
+# prune the reflog as well. It is important to prevent corruption that we do
+# leave *some* grace period in all of these (see the note in docs for git gc
+# for more about this).
+RUN git config --global gc.auto 20000
+RUN git config --global gc.pruneExpire 1.hour.ago
+RUN git config --global gc.reflogExpire 1.hour.ago
+RUN git config --global gc.reflogExpireUnreachable 1.hour.ago
```

Comment:
those are quite aggressive values 😄 (defaults are in weeks/months)

Reply:
Yeah! Those defaults make a lot of sense for interactive use! Like, I've totally needed to go back to commits that aren't part of a branch a day or two later. `git reflog` is amazing! But this machine won't.